public/app/plugins/datasource/tempo/NativeSearch.tsx

```diff
@@ -41,25 +42,38 @@ Prism.languages[PRISM_LANGUAGE] = tokenizer;
 const NativeSearch = ({ datasource, query, onChange, onBlur, onRunQuery }: Props) => {
   const languageProvider = useMemo(() => new TempoLanguageProvider(datasource), [datasource]);
   const [hasSyntaxLoaded, setHasSyntaxLoaded] = useState(false);
-  const [autocomplete, setAutocomplete] = useState<{
-    serviceNameOptions: Array<SelectableValue<string>>;
-    spanNameOptions: Array<SelectableValue<string>>;
+  const [autocomplete] = useState<{
+    selectedServiceName: SelectableValue<string> | undefined;
+    selectedSpanName: SelectableValue<string> | undefined;
   }>({
-    serviceNameOptions: [],
-    spanNameOptions: [],
+    selectedServiceName: undefined,
+    selectedSpanName: undefined,
   });
 
   useEffect(() => {
     const fetchAutocomplete = async () => {
       await languageProvider.start();
-      const serviceNameOptions = await languageProvider.getOptions('service.name');
-      const spanNameOptions = await languageProvider.getOptions('name');
       setHasSyntaxLoaded(true);
-      setAutocomplete({ serviceNameOptions, spanNameOptions });
     };
     fetchAutocomplete();
   }, [languageProvider]);
 
+  const debouncedFetchServiceNameOptions = debounce(
```

Comment:
Did not try it yet but I think this needs `useCallback` or something like that doesn't it? Otherwise this will create a debounced function on every rerender with it's own timer, I would assume we need only one static function for it so that rerenders does not reset the timer (also I would assume the old debounced function could still fire anyway as there is no cleanup for them)

Reply:
I'll test that out although the other components using `AsyncSelect` with a debounced function don't so it might be okay as is.

Reply:
I had to use `useMemo` in this case to get it working properly. This blog post has some more info on debouncing in React: https://dmitripavlutin.com/react-throttle-debounce/